ledger-core/v2/vanilla/synckit/closed_channel.go

```diff
@@ -5,12 +5,26 @@
 
 package synckit
 
-func ClosedChannel() <-chan struct{} {
+import "github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+
+type SignalChannel = <-chan struct{}
+
+type ClosableSignalChannel = chan struct{}
+
+func ClosedChannel() SignalChannel {
 	return closedChan
 }
 
-var closedChan = func() chan struct{} {
-	c := make(chan struct{})
+var closedChan = func() SignalChannel {
+	c := make(ClosableSignalChannel)
 	close(c)
 	return c
 }()
+
+func SafeClose(c ClosableSignalChannel) (err error) {
+	defer func() {
+		err = throw.R(recover(), err)
+	}()
+	close(c)
```

Comment:
Generally you don't want to use recover() in Go. In this case it basically means that you are writing in the channel which state you are unsure of. Please consider rewriting the code without using recover().

Reply:
I do want to do it. Won't reconsider, sorry ;)